### ISSUE TITLE
PromQL doesn't support =~ on metric name

### ISSUE THREAD

**Issue by 0neSe7en** [GreptimeTeam/greptimedb#5182]
### What type of bug is this?

Incorrect result

### What subsystems are affected?

Frontend

### Minimal reproduce step

```
tql eval(1733995543,1734595483, '10m') {__name__=~"metric_a|metric_b"}
```

### What did you expect to see?

query from metric_a and metric_b

### What did you see instead?

```
Table not found: greptime.metrics.metric_a|metric_a
```

### What operating system did you use?

doesn't matter

### What version of GreptimeDB did you use?

0.11.0

**Comment by killme2008** [comment]
Yes, the current implementation cannot retrieve multiple metrics in a single query.

**Comment by waynexia** [comment]
This limitation is documented here https://docs.greptime.com/user-guide/query-data/promql#selector, I add a validation for this case in #5191, thanks for reporting this
